File: sdks/java/src/main/java/io/licensemanager/sdk/LicenseManagerException.java
package io.licensemanager.sdk;

public class LicenseManagerException extends Exception {
    private final int statusCode;
    private final String responseBody;

    public LicenseManagerException(int statusCode, String responseBody) {
        super("LicenseManager API error " + statusCode + ": " + responseBody);
        this.statusCode   = statusCode;
        this.responseBody = responseBody;
    }

    public int getStatusCode()     { return statusCode; }
    public String getResponseBody(){ return responseBody; }
}

File: sdks/php/src/LicenseManagerException.php
<?php

declare(strict_types=1);

namespace LicenseManager\Sdk;

use RuntimeException;

/**
 * Exceção lançada quando a API retorna um erro HTTP.
 */
class LicenseManagerException extends RuntimeException
{
    public function __construct(
        private readonly int $statusCode,
        private readonly string $responseBody
    ) {
        parent::__construct(
            sprintf('LicenseManager API error %d: %s', $statusCode, $responseBody)
        );
    }

    public function getStatusCode(): int
    {
        return $this->statusCode;
    }

    public function getResponseBody(): string
    {
        return $this->responseBody;
    }
}
